source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -376,6 +384,24 @@ namespace PlayFab
             playFabEmitRequest->stdCallback(playFabEmitRequest->event, response);
         }
     }
+
+    bool PlayFabEventPipeline::TryGetBatchOutOfFlight(void* customData, std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>* batchReturn)
+    {
+        // LOCK batchesInFlight mutex
+        std::unique_lock<std::mutex> lock(inFlightMutex);
+        auto iter = this->batchesInFlight.find(customData);
+        if (iter == this->batchesInFlight.end())
+        {
+            // not finding the batch in the queue is a bug
+            LOG_PIPELINE("Untracked batch was returned to EventsAPI.WriteEvents callback"); // normally this never happens
+            return false;
+        }
+
+        *batchReturn = std::move(iter->second);
+        this->batchesInFlight.erase(iter);
+        return true;
+        // UNLOCK batchesInFlight
```

Comment:
I'm not super familiar with how your team likes these UNLOCK comments, but it's noting that you also unlock at the return false line on 397 (which is good I'm just not sure if you were missing a comment)